tcms/bugs/tests/test_views.py

```diff
@@ -209,3 +209,65 @@ def test_record_changes(self):
             comments.last().comment,
             "Summary: %s -> %s\n" % (old_summary, new_summary)
         )
+
+
+class TestAddComment(LoggedInTestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        user_should_have_perm(cls.tester, 'django_comments.add_comment')
+        user_should_have_perm(cls.tester, 'bugs.view_bug')
+
+        cls.bug = BugFactory()
+
+        cls.url = reverse('bugs-comment')
+        cls.post_data = {
+            'bug': cls.bug.pk,
+            'text': 'A comment text',
+        }
+
+    def test_add_comment(self):
```

Comment:
can turn this into a scenario where the post data is invalid, e.g. a negative bug id. Should display errors and not add any comments. That will cover the jump from L192 to 207.

Reply:
Turning this into a scenario where the post data is invalid, I bumped into an `UnboundLocalError` arising because in this scenario `bug` is never assigned before it is referenced on L207.

Should I handle that path in this PR?

Reply:
You can't. If the bug ID is invalid then there's no way to properly redirect to the page and show any form errors. 

Looking at BugCommentForm there isn't anything else there to easily make it invalid.  This scenario doesn't make sense here. Will merge as-is.